### PR TITLE
handle zero-result queries, specify index/type, add stub of rspec

### PR DESCRIPTION
**Pull request by kurtado**
Attempting to aggregate several PRs, feature requests and issues:

https://github.com/elasticsearch/logstash-contrib/pull/57
https://github.com/elasticsearch/logstash-contrib/issues/95
https://github.com/elasticsearch/logstash-contrib/pull/96

with help and input from @greem @sisutcliffe @bleskes et al.

### CODE REVIEW

File: lib/logstash/filters/elasticsearch.rb [hunk 1]
```diff
@@ -38,8 +38,14 @@ class LogStash::Filters::Elasticsearch < LogStash::Filters::Base
 
   # Elasticsearch query string
   config :query, :validate => :string
+ 
+  # Index to search in
+  config :index, :validate => :string, :default => ""
```
Comment (jordansissel) [inline lib/logstash/filters/elasticsearch.rb]: A default value of empty string seems confusing to expose in the documentation. I'm OK with the default *behavior* being "all indices" but the documentation should just state the default behavior. Showing, in the docs, 'the default value is ""' is confusing to me.

I recommend having no `:default => ...` and just declaring the default behavior.

**Comment by purbon** [comment]
Is this PR still meaningful? it just need so few work to get it done :+1: , what is your proposal plan of action?

**Comment by talevy** [comment]
@kurtado plan to push through with this PR? otherwise, we can close this, and rework it in a followup. let us know what you decide!

**Comment by jsvd** [comment]
the code has diverged from master anyway, so I'm closing until interest pops up again

**Pull request closed without merging**